qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,342 @@
+        ansatz: Optional[QuantumCircuit] = None,
+        threshold: float = 1e-5,
+        max_iterations: Optional[int] = None,
+        adapt_gradient: Optional[GradientBase] = None,
+        initial_point: Optional[np.ndarray] = None,
+        expectation: Optional[ExpectationBase] = None,
+        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
+        excitation_pool: List[Union[OperatorBase, QuantumCircuit]] = None,
+    ) -> None:
+        """
+        Args:
+            solver: a factory for the VQE solver employing a UCCSD ansatz.
+            ansatz: A parameterized circuit used as Ansatz for the wave function.
+            threshold: the energy convergence threshold. It has a minimum value of 1e-15.
+            max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            adapt_gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
+            initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+            expectation: The Expectation converter for taking the average value of the
+                Observable over the ansatz state function.
+            quantum_instance: Quantum Instance or Backend
+            excitation_pool: An entire list of excitations.
+        """
+        validate_min("threshold", threshold, 1e-15)
+
+        if adapt_gradient is None:
+            adapt_gradient = Gradient(grad_method="param_shift")
+        self._threshold = threshold
+        self.solver = solver
+        self.initial_point = initial_point
+        self._max_iterations = max_iterations
+        self._adapt_gradient = adapt_gradient
+        self._excitation_pool = excitation_pool
+        self._tmp_ansatz = ansatz
+        self.expectation = expectation
+        self.quantum_instance = quantum_instance
+        self._excitation_list: List[OperatorBase] = []
+
+    @property
+    def initial_point(self) -> Optional[np.ndarray]:
+        """Returns initial point."""
+        return self.solver.initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: Optional[np.ndarray]) -> None:
+        """Sets initial point."""
+        self.solver.initial_point = initial_point
+
+    def _compute_gradients(
+        self,
+        theta: List[float],
+        operator: OperatorBase,
+    ) -> List[Tuple[float, PauliSumOp]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters
+            operator: operator whose gradient needs to be computed
+        Returns:
+            List of pairs consisting of gradient and excitation operator.
+        Raises:
+            AlgorithmError: If `quantum_instance` is not provided.
+        """
+        res = []
+        # compute gradients for all excitation in operator pool
+        if self.quantum_instance is None:
```

Comment:
I think this should be taken from `self.solver` instead. However, the `is None` check should still be done